enferno/admin/models/DynamicField.py

```
@@ -0,0 +1,425 @@
+    @classmethod
+    def get_valid_components(cls, field_type: str) -> list:
+        """Get valid UI components for a field type"""
+        return cls.COMPONENT_MAP.get(field_type, [])
+
+    def validate_field(self):
+        """Validate field configuration"""
+        if not self.name:
+            return  # Skip validation if name is not set yet
+
+        if not self.name.isidentifier():
+            raise ValueError("Field name must be a valid Python identifier")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)